connectionmanager/connectionmanager.go

```diff
@@ -211,9 +217,11 @@ func setReconnecting(proto ws.ProtoType, v bool) bool {
 	return v
 }
 
-func CancelReconnection(proto ws.ProtoType) {
+func CancelReconnection(proto ws.ProtoType) bool {
 	maxWaitSeconds := 8
-	cancelReconnectChan[proto] <- true
+	go func() {
+		cancelReconnectChan[proto] <- true
```

Comment:
Suggestion:
What about just giving the channel a buffer size?

Reply:
How will a buffered channel prevent the deadlock? Won't it be the same problem? The base problem is that during the reconnection there is a for loop (previously forever, now up to 10 attempts) where this channel might be read but the loop can also return on errors or on success without reading the channel (hence this "draining" I also wrote there, line 138).